### PR TITLE
[carbon-graphs] Improve reflow's performance.

### LINKED ISSUE #35: Reflow is taking long time to load the graph
**What part of Carbon does this bug correlate to?**
General graph logic:

* [X]  Graph attributes (axes, legends, regions...etc.)
* [ ]  Other (please specify): 

Specific graph logic:

* [ ]  Bar graph
* [ ]  Bubble graph
* [ ]  Gantt chart
* [ ]  Line graph
* [ ]  Paired result
* [ ]  Pie chart
* [ ]  Scatter plot
* [ ]  Timeline

**Describe the bug**
When we try to load a graph during reflow it takes a considerable amount of time than the time taken to create the graph. 

**Expected behavior**
the time to load the graph for that particular timeframe during reflow should be less than the time taken to create the graph since we are not destroying or the existing graph object and we are just updating the graph.
It took for hardly 4 seconds to load the graph
Screenshot 2020-10-19 at 2 40 09 PM

Whereas it took around 7 seconds to update graph during panning
Screenshot 2020-10-19 at 2 44 49 PM

### PR DESCRIPTION
**Pull request by pranav300**
### Summary

This PR fixes reflows performance by allowing users to send in multiple datasets(key-value pairs) to the reflow function using an array panData, inside the object to be sent to the reflow function.

To maintain passivity I have added a new function `reflowMultipleDatasets`, which can later be renamed to `reflow` in the next major version bump and remove the current `reflow` in it as well.

#### graphData Structure for `reflowMultipleDatasets`
```jsx
graphData: {
  panData: [
    {
      Key: 'uid_1',
      values: [],
    },
   {
      Key: 'uid_2',
      values: [],
    },
  ],
  evenline: [],
  xLabel: '',
  yLabel: '',
  y2Label: '',
}
```

Closes #35 

### Deployment Link

https://terra-graphs-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra
@cerner/carbon

### CODE REVIEW

**Comment by AshishMotanamGurunadham** [comment]
@pranav300 Can you please add the documentation for the new method added.

**Comment by NarasimhaShenoi** [comment]
Hi All, when this can be merged. can you please provide an ETA for the same?

File: packages/carbon-graphs/src/js/controls/Graph/Graph.js [hunk 1]
```diff
@@ -422,68 +422,129 @@ class Graph extends Construct {
   /**
      * Updates the graph axisData and content.
      *
-     * @param {Array} graphData - Input array that holds updated values, key and labels
+     * @param {Object} graphData - Input object that holds updated values and key pair and labels
      * @returns {Graph} - Graph instance
      */
   reflow(graphData) {
-    let position;
+    console.warn('reflow is now deprecated and will be removed in future Major Version Bump. Please use reflowMultipleDatasets');
```
Comment (sdadn) [inline packages/carbon-graphs/src/js/controls/Graph/Graph.js:429]: nit: reflow is deprecated and will be removed a future major release. Please use reflowMultipleDatasets instead.

File: packages/carbon-graphs/tests/unit/controls/Bar/BarPanningMultipleDatasets-spec.js [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+          shape: SHAPES.SQUARE,
+          value: '2016-06-03T12:00:00Z',
+        },
+      ];
+      axisData.pan = { enabled: true };
+      const input = getInput(valuesTimeSeries, false, false);
+      graphDefault = new Graph(axisData);
+      graphDefault.loadContent(new Bar(input));
+    });
+    it('Check if clamp is false if pan is enabled', () => {
```
Comment (sdadn) [inline packages/carbon-graphs/tests/unit/controls/Bar/BarPanningMultipleDatasets-spec.js:62]: nit: checks

File: packages/carbon-graphs/tests/unit/controls/Bar/BarPanningMultipleDatasets-spec.js [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+      ];
+      axisData.pan = { enabled: false };
+      const input = getInput(valuesTimeSeries, false, false);
+      graphDefault = new Graph(axisData);
+      graphDefault.loadContent(new Bar(input));
+    });
+    it('Check if clamp is true if pan is disabled', () => {
+      expect(graphDefault.scale.x.clamp()).toEqual(true);
+    });
+    it('DatelineGroup translates properly after some delay when panning is disabled', (done) => {
```
Comment (sdadn) [inline packages/carbon-graphs/tests/unit/controls/Bar/BarPanningMultipleDatasets-spec.js:337]: nit: translates DatelineGroup properly

File: packages/carbon-graphs/tests/unit/controls/Bar/BarPanningMultipleDatasets-spec.js [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+          shape: SHAPES.SQUARE,
+          value: '2016-06-03T12:00:00Z',
+        },
+      ];
+      axisData.pan = { enabled: false };
+      const input = getInput([], false, false);
+      graphDefault = new Graph(axisData);
+      graphDefault.loadContent(new Bar(input));
+    });
+    describe('when legend hold values', () => {
```
Comment (sdadn) [inline packages/carbon-graphs/tests/unit/controls/Bar/BarPanningMultipleDatasets-spec.js:370]: nit: when legend holds values

File: packages/carbon-graphs/tests/unit/controls/Bubble/Bubble/BubblePanningMultipleDatasets-spec.js [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+          shape: SHAPES.SQUARE,
+          value: '2016-06-03T12:00:00Z',
+        },
+      ];
+      axisData.pan = { enabled: true };
+      const input = getInput(valuesTimeSeries, false, false);
+      graphDefault = new Graph(axisData);
+      graphDefault.loadContent(new Bubble(input));
+    });
+    it('Check if clamp is false if pan is enabled', () => {
```
Comment (sdadn) [inline packages/carbon-graphs/tests/unit/controls/Bubble/Bubble/BubblePanningMultipleDatasets-spec.js:69]: nit: checks

File: packages/carbon-graphs/tests/unit/controls/Bubble/MultipleDataset/BubblePanningMultipleDatasets-spec.js [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+          shape: SHAPES.SQUARE,
+          value: '2016-06-03T12:00:00Z',
+        },
+      ];
+      axisData.pan = { enabled: true };
+      const input = getInput(valuesTimeSeries, false, false);
+      graphDefault = new Graph(axisData);
+      graphDefault.loadContent(new BubbleMultipleDataset(input));
+    });
+    it('Check if clamp is false if pan is enabled', () => {
```
Comment (sdadn) [inline packages/carbon-graphs/tests/unit/controls/Bubble/MultipleDataset/BubblePanningMultipleDatasets-spec.js:70]: nit: checks

File: packages/carbon-graphs/tests/unit/controls/Bubble/MultipleDataset/BubblePanningMultipleDatasets-spec.js [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+      expect(graphDefault.scale.x.clamp()).toEqual(false);
+    });
+    it('DatelineGroup translates properly when panning is enabled', (done) => {
```
Comment (sdadn) [inline packages/carbon-graphs/tests/unit/controls/Bubble/MultipleDataset/BubblePanningMultipleDatasets-spec.js:73]: nit: translates DatelineGroup properly

File: packages/carbon-graphs/tests/unit/controls/Bubble/MultipleDataset/BubblePanningMultipleDatasets-spec.js [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+          expect(axisLabelY.querySelector('text').textContent).toBe(
+            'updated yLabel',
+          );
+          expect(axisLabelY2.querySelector('text').textContent).toBe(
+            'updated y2Label',
+          );
+        });
+      });
+    });
+    it('Dynamic Data is not updated when key does not match', () => {
```
Comment (sdadn) [inline packages/carbon-graphs/tests/unit/controls/Bubble/MultipleDataset/BubblePanningMultipleDatasets-spec.js:227]: nit: does not update dynamic data when key does not match

**Comment by joelsamthomas** [comment]
@pranav300 any ETA on this. We are planning to consume this version once this gets released

**Review by AshishMotanamGurunadham — APPROVED** [review]

**Review by sdadn — APPROVED** [review]

**Review by mraplumb — APPROVED** [review]

**Pull request merged**